### DISCUSSION TITLE
Welcome to OTP-designer-jquery ideas club

### DISCUSSION THREAD

**Discussion by HichemTab-tech** [HichemTab-tech/OTP-designer-jquery#4] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ADProperant** [comment]
Is there any possibility to enable the paste command into OPT fake input? If we use right button on mouse o double tap on mobile device than no suggestions appears to paste the OTP code.

**Reply by HichemTab-tech** [reply]
Well, there is actually **paste** functionality (using `Ctrl+V`) but yes, we can bring up the context menu, to choose **paste** using the mouse.
THANKS

**Reply by ADProperant** [reply]
Unfortunately if we bring up the context menu with mouse there isn't the paste voice. I trying this in your demo page.
Of course ctrl+v works, becouse your code it listener that action, but as a fake input the paste from mouse not works in this OTP.

**Reply by ADProperant** [reply]
Same if we use a mobile device and long press on the fake OTP to show the paste voice (that not works as above).

**Reply by ADProperant** [reply]
I see just now that you had open a issue topic ;) I follow up

**Reply by HichemTab-tech** [reply]
Hi @ADProperant ,

I wanted to let you know that I've completed the branch for issue #12 regarding enabling the paste command into the OTP fake input. Regarding your suggestion about showing the browser context menu, unfortunately, that approach isn't feasible at the moment, as the context menu meant for inputs can't be shown in other places like fake inputs.

Instead, I implemented a workaround solution by adding a new custom context menu. This custom context menu is editable by developers, allowing them to add custom options or use the default ones, including the paste functionality. This approach works well in desktop browsers, provided users accept clipboard permissions. However, on Android phones, there might be limitations, especially on insecure domains (non-HTTPS), due to restrictions imposed by Google Chrome. In such cases, users can still use the old method of pasting using the Android keyboard.

Overall, the new solution is effective for desktop environments and provides flexibility for developers to customize the context menu as needed.

You can check out the details in branch #12. Let me know if you have any further propositions.
Best regards,
@HichemTab-tech
